docs/content/direct/multi-wec-aggregated-status-proposal.md

```diff
@@ -0,0 +1,121 @@
+
+The enhancement requires updating both the `BindingPolicy` and `Binding` API types.
+
+## Controller Behavior
+
+If the user sets `wantMultiWECReportedState: true`, the controller will:
+
+1. Identify the WECs selected for the workload object.
+2. Retrieve the reported status from each selected WEC.
+3. Determine whether to apply singleton copying or aggregation:
+   - One WEC → copy status
+   - More than one WEC → aggregate status
+4. Write the resulting status into the WDS workload object.
+
+The existing rules for singleton status return remain unchanged.
+
+## Aggregation Approach
+
+Aggregation rules depend on the workload kind.
+
+### Workload Kinds with Argo CD Health Rules
+
+For kinds with built-in Argo CD health assessment:
+- Identify the subset of fields used by Argo CD.
+- Aggregate readiness-related numeric fields using minimum.
+- Aggregate conditions using three-value logic.
+- Ignore fields not used by Argo CD’s health evaluator.
+
+### Other Workload Kinds
+
+- Numeric fields: use minimum across WECs.
+- Boolean fields: false if any false, true only if all true.
+- Conditions: three-value logic for truth value, latest timestamp, and reason/message from the newest entry.
+- List/map fields: aggregate only when structurally compatible.
+
+These rules aim to provide a practical summarization without inferring unsupported semantics.
+
+## Data Flow
+
+1. WECs publish status into their respective ExecutionSpaces.
+2. The status controller watches these updates.
+3. For each workload in the WDS, the controller collects the relevant per-WEC statuses.
+4. The controller computes the aggregated output.
+5. The controller updates the `.status` field of the WDS object.
+
+The existing `CombinedStatus` objects continue to store per-WEC details and remain unaffected.
+
+## Validation Plan
+
+The following cases will be validated:
+
+- **Singleton fallback**: One WEC selected → `.status` matches the WEC's reported status.
+- **Multi-WEC aggregation** for:
+  - Deployments (varied readiness across WECs)
+  - StatefulSets
+  - DaemonSets
+  - Jobs (handling of `active`, `succeeded`, `failed`)
+  - Arbitrary workload kinds without Argo CD rules
+- **Inconsistent fields**:
+  - Incompatible list/map structures
+  - Mixed boolean values
+  - Divergent condition timestamps
+- **Argo CD integration**:
+  - Aggregated `.status` leads to expected health assessments.
+
+## Risks and Mitigations
+
+- **Misinterpretation of numeric fields**  
+  Not all numeric fields represent readiness.  
+  *Mitigation:* Aggregate only fields covered by Argo CD rules for known kinds; use minimal general aggregation for unknown kinds.
+
+- **Large numbers of WECs**  
+  Aggregation cost and update frequency may increase.  
+  *Mitigation:* Continue relying on existing rate-limiting and batching.
+
+- **Inconsistent field shapes across WECs**  
+  Some fields cannot be merged safely.  
+  *Mitigation:* Omit fields that differ structurally.
```

Comment:
Now I am confused about what this document means by "field". Mostly it seems to implicitly assume that a "field" holds an atomic value. But here we are talking about a field that holds a structured value.